### PR TITLE
Add ability to specify specific line or line blocks

### LINKED ISSUE #3: Add Support for line numbers
I was wondering whether you were planning support for line numbers as well as filenames ?  
If so, I might take a shot at implementing this and sending in a PR, because I would love to have this for a project of mine.

The idea would be to allow for line numbers after the filename:
```markdown
Code block with single quotes because backticks mess up formatting:
'''js file=./hello.js:5-10
'''

Should separate linenumbers be allowed as well ?
This might not really be useful, but why not (if not too complicated).
What do you think ?
'''js file=./hello.js:1-5;10;13-17
'''

```

Let me know if you would be interested in such a PR or not!

### PR DESCRIPTION
**Pull request by ostollmann**
Close #3 

This adds the possibility to specify specific lines to be imported:

````
```js file=./__fixtures__/say-hi.js#L2
```
````

````
```js file=./__fixtures__/say-hi.js#L1-L2
```
````

Note: couldn't get yarn to work, but tests passed using npm.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -20,9 +20,23 @@ function codeImport(options = {}) {
...
+      }
+      const filePath = res.groups.path
+      const fromLine = res.groups.from ? parseInt(res.groups.from) : undefined
+      const toLine = res.groups.to ? parseInt(res.groups.to) : undefined
       const fileAbsPath = path.resolve(file.dirname, filePath);
 
+      const extractLines = (content, fromLine, toLine) => {
+        if (fromLine === undefined && toLine === undefined) { return content }
+        if (toLine === undefined) toLine = fromLine
+        const lines = content.split(/\r?\n/)
```
Comment (kevin940726) [inline index.js]: Maybe we could split it with `os.EOL`?

File: index.js [hunk 2]
```diff
@@ -20,9 +20,23 @@ function codeImport(options = {}) {
         continue;
       }
 
-      const filePath = fileMeta.slice('file='.length);
+      const res = /^file=(?<path>[^#]+)(?:#L(?<from>\d+))?(?::L(?<to>\d+))?$/.exec(fileMeta)
```
Comment (kevin940726) [inline index.js]: This will fail to match if the path itself contains `#` somewhere. Could we adjust this so that only the last `#` is accounted?

**Review by kevin940726 — CHANGES_REQUESTED** [review]
Thank you for creating this PR!

What do you think if we also support `#L3:` pattern to slice the file from line 3 to the very end? I think it could be a very common use case when all the user wants is to strip out the import statements.

Just being curious, where do you know of this pattern `#Ls:Ln`? I feel like we can try to follow what GitHub does and use `-` to replace `:`. `#L3-` is also slightly more readable to me too.

File: index.js (hunk 2 above)
Comment (ostollmann) [reply index.js]: Good catch. Fixed.

Note I also changed the test file to `say-#-hi.js` to test for this.

**Comment by ostollmann** [comment]
> Thank you for creating this PR!
> 

Np. I want to use it :)

> What do you think if we also support `#L3:` pattern to slice the file from line 3 to the very end? I think it could be a very common use case when all the user wants is to strip out the import statements.

That's a good idea. I added the ability to do both:
- preceding lines: `js file=./__fixtures__/say-#-hi.js#-L2`
- following lines: `js file=./__fixtures__/say-#-hi.js#L2-`

> 
> Just being curious, where do you know of this pattern `#Ls:Ln`? I feel like we can try to follow what GitHub does and use `-` to replace `:`. `#L3-` is also slightly more readable to me too.

I wanted to do what Github does and thought they used `?:?`. Should have checked. Changed to `?-?` now.

**Comment by ostollmann** [comment]
I also added this to the readme.

**Comment by kevin940726** [comment]
I just want to let you know that I've been a little bit busy lately, and I'll get back to it as soon as I have some time. Thank you for all the good work and your patience :)

File: README.md [hunk 3]
```diff
@@ -48,6 +48,24 @@ console.log('Hello remark-code-import!');
...
+```js file=./say-hi.js#L3
+```
+
+```js file=./say-hi.js#L3-L6
+```
+
+```js file=./say-hi.js#L3-
+```
+
+```js file=./say-hi.js#-L6
```
Comment (kevin940726) [inline README.md:63]: I don't think this is useful as it's the same as `#L1-L6` and it complicates the rule a little bit. I believe we should remove it, WDYT?

File: index.js [hunk 4]
```diff
@@ -20,9 +21,24 @@ function codeImport(options = {}) {
...
+      if (!res || !res.groups || !res.groups.path) {
+        throw new Error(`Unable to parse file path ${fileMeta}`)
+      }
+      const filePath = res.groups.path
+      const hasDash = !! res.groups.dash
+      const fromLine = res.groups.from ? parseInt(res.groups.from) : undefined
+      const toLine = res.groups.to ? parseInt(res.groups.to) : undefined
       const fileAbsPath = path.resolve(file.dirname, filePath);
 
+      const extractLines = (content, fromLine, toLine) => {
```
Comment (kevin940726) [inline index.js]: Prefer to move this function outside the `codeImport` function and make it pure (side-effect-free) so that it's easier to reason about.

**Review by kevin940726 — CHANGES_REQUESTED** [review]
Good work!

There are some styling issues and tooling issues, probably has something to do with yarn berry not working correctly on your machine. I can handle that after we merge this though.

File: README.md (hunk 3 above)
Comment (ostollmann) [reply README.md:63]: Indeed. I removed this and added a test case to check for it: https://github.com/o1iver/remark-code-import/blob/0051e274e9340b0be4b9db5e78641bc243e165d7/test.js#L65.

File: index.js (hunk 4 above)
Comment (ostollmann) [reply index.js]: Not quite sure what you mean with pure or what side-effect it was creating; it was always pure, even it defined inline (except for `hasDash` which it wasn't modifying). The args shadow the local vars so it's not even a closure. I extracted `extractLines` to the top level though if that's what you meant.

**Review by kevin940726 — APPROVED** [review]
Thank you! There are some minor styling preference of mine. I'll update them. 👍

File: index.js (hunk 4 above)
Comment (kevin940726) [reply index.js]: Yeah, I just prefer it to be outside the main function so that it's a little bit more clear for me to review 😅.

**Pull request merged**